README.md

```diff
@@ -346,21 +352,130 @@ const perfume = new Perfume({
+
+```typescript
+// Marking the start of the step
+const ScreenA = () => {
+  const handleNavigation = () =>  {
+    ... // Navigation logic
+    // Mark when navigating to screen B
+    markStep('navigate_to_screen_B');
+  }
+
+  ...
+
+  return (
+    <>
+      <Button onPress={handleNavigation} />
+    </>
+  );
+}
+
+// Marking the end of the step
+const ScreenB = () => {
+  const { viewer } = fetch("http://example.com/userInfo")
+                     .then((response) => response.json())
+                     .then((data) => data);
+
+  const {name} = viewer.userProperties;
+
+  useEffect(() => {
+    if (name) {
+      // Mark when data is ready for screen B
+      markStep('loaded_screen_B');
+    }
+  }, [name])
+
+  ...
+}
+```
+
+#### Defining Steps
+In order for Perfume to be able to track metrics for Steps, we need to configure the steps and provide them when initializing Perfume.
+
+Below you can find an example of how to do this.
+
+``` typescript 
+export const steps = {
+  load_screen_A: {
+    threshold: ThresholdTier.quick,
+    marks: ['navigate_to_screen_A', 'loaded_screen_A'],
+  },
+  load_screen_B: {
+    threshold: ThresholdTier.quick,
+    marks: ['navigate_to_screen_B', 'loaded_screen_B'],
+  },
+};
+
+new Perfume ({ steps });
+
+``` 
+
+#### MarkStep
+`markStep` is the function used to start and end steps in applications.
+
+For example, if we wanted to mark the beginning of load_screen_B step above, we would add in `markStep('navigate_to_screen_B')` to our code.
+
+#### `enableNavigationTracking`
+
+`enableNavigationTracking` is a boolean in the config that will tell Perfume to take into account page navigation changes or not. By default this is `true`, but it can be set to false if needed. 
+
+The purpose of this feature is to only account for active steps that the user is working on. The feature will remove any inactive or 'stale' steps that are not currently in progress. 
+
+Stale steps can be created by navigating away from a page before it fully loads, this would cause the start mark to be triggered, but the end mark to not be called. This would affect the `active` steps being returned to `onMarkStep` as well as would create incorrect data if we returned back to the end mark much later than expected. 
+
+`enableNavigationTracking` works together with the `incrementUjNavigation` function. The `incrementUjNavigation` function is to be called anytime there is a navigation change in your application. Below is an example for how this would work in a React Application:
+
+``` typescript 
+import { useLocation } from 'react-router-dom';
+
+const MyComponent = () => {
+  const location = useLocation()
+
+  React.useEffect(() => {
+    // runs on location, i.e. route, change
+    incrementUjNavigation();
```

Comment:
Missed the import in this docs, also `incrementUjNavigation` feels a bit odd as name. What other options we have?

Reply:
Let's explore that in the follow up PR.